crates/alkahest/src/util/action.rs

```diff
@@ -206,3 +242,120 @@ impl Action for SpawnRouteAction {
         false
     }
 }
+
+pub struct FollowAction {
+    route_ent: Entity,
+    traverse_from: Option<Entity>,
+}
+
+impl FollowAction {
+    pub fn new(route_ent: Entity, traverse_from: Option<Entity>) -> Self {
+        Self {
+            route_ent,
+            traverse_from,
+        }
+    }
+}
+
+impl Action for FollowAction {
+    fn start(&mut self, resources: &AppResources) {
+        let camera = resources.get::<Camera>();
+        let mut maps = resources.get_mut::<MapList>();
+
+        if let Some(map) = maps.current_map_mut() {
```

Comment:
This if block is quite big, i'd suggest using a let else binding here to use an early return instead